agent/config/builder.go

```diff
@@ -1021,7 +1024,7 @@ func (b *builder) build() (rt RuntimeConfig, err error) {
 		NodeMeta:                          c.NodeMeta,
 		NodeName:                          b.nodeName(c.NodeName),
 		ReadReplica:                       boolVal(c.ReadReplica),
-		PeeringEnabled:                    boolVal(c.Peering.Enabled),
+		PeeringEnabled:                    boolValWithDefault(c.Peering.Enabled, true),
```

Comment:
This one should probably also have the same `agent/config/default.go` treatment (even though the field only appears to be used by servers currently). We could also enable it for dev mode by default.